map/views.py

```diff
@@ -1292,6 +1292,55 @@ def get_tile(self, req, ver, zoom, x, y,
     
         if get_images:
             return rimgs
+
+        if "lslga" in req.GET:
+
+            from PIL import Image, ImageDraw
+            img = Image.open(tilefn)
+
+            ra, dec = wcs.crval
+            pixscale = wcs.pixel_scale()
+
+            width, height = img.size
+
+            ralo = ra - ((width / 2) * pixscale / 3600 / np.cos(np.deg2rad(dec)))
+            rahi = ra + ((width / 2) * pixscale / 3600 / np.cos(np.deg2rad(dec)))
+            declo = dec - ((height / 2) * pixscale / 3600)
+            dechi = dec + ((height / 2) * pixscale / 3600)
+
+            from map.cats import query_lslga_radecbox
+            galaxies = query_lslga_radecbox(ralo, rahi, declo, dechi)
+
+            for r in galaxies:
+
+                RA, DEC = r.ra, r.dec
+                RAD = r.radius_arcsec
+                AB = r.ba
+                PA = r.pa
+
+                major_axis_arcsec = RAD * 2
+                minor_axis_arcsec = major_axis_arcsec * AB
+
+                overlay_height = int(major_axis_arcsec / pixscale)
+                overlay_width = int(minor_axis_arcsec / pixscale)
+
+                overlay = Image.new('RGBA', (overlay_width, overlay_height))
+                draw = ImageDraw.ImageDraw(overlay)
+                box_corners = (0, 0, overlay_width, overlay_height)
+                draw.ellipse(box_corners, fill=None, outline=(0, 0, 255), width=3)
+
```

Comment:
@ver09934 @dstndstn I'm getting an `ellipse() got an unexpected keyword argument 'width'` error when querying the cutout server with this option, and indeed `width` does not appear to be a valid argument--
https://pillow.readthedocs.io/en/3.1.x/reference/ImageDraw.html?highlight=ellipse#PIL.ImageDraw.PIL.ImageDraw.Draw.ellipse

What am I missing?

Reply:
I think it has something to do with the `Pillow` version being used. It's working for me locally using `Pillow==6.0.0`. Looking at the [release notes](https://pillow.readthedocs.io/en/stable/releasenotes/5.3.0.html#added-line-width-parameter-to-rectangle-and-ellipse-based-shapes), it seems like the `width` option was added in version `5.3.0`, so I assume the version running on the server is something prior to that. If it's straightforward to upgrade the `Pillow` version to `5.3.0` or above, that seems like the simplest option. Otherwise, the `width` argument could be removed from the `ellipse()` call, although this might make the ellipses more difficult to see in some cases.

Reply:
@ziyaointl could you update the version of `pillow` in the viewer container?

Reply:
@moustakas @ver09934 I updated pillow in the dev version of the viewer to 6.2.0. Let me know if it works, and I can change the container version for the production version too.

Reply:
Thanks @ziyaointl.  

I'm not sure what's going.  This now runs (in `viewer-dev`, as expected, but not the production `viewer`), but doesn't generate the ellipse--
http://legacysurvey.org/viewer-dev/jpeg-cutout?ra=195.0338&dec=27.9769&layer=dr8&size=1500&lslga
Is this the expected behavior, @ver09934?

By contrast, your inspection tools is rendering the ellipse but maybe you're drawing on top of the cutout requested from the viewer?
http://lslga-inspect.legacysurvey.org/lslga?id=818201&drawname&drawscale&ellipse&layer=dr8&annotation=Data&width=1500&height=1500&fontsize=30&ellipsewidth=5

Actually, looking at the code, I think I have a bug and I should have 
`if req.GET.get('lslga') == ''` 
rather than 
`if req.GET == 'lslga'`--
https://github.com/legacysurvey/decals-web/blob/master/map/views.py#L1322-L1329

Reply:
Right now the viewer-dev link you gave is throwing `local variable 'AB' referenced before assignment`. I think #64 might have just missed another instance where
`if req.GET == 'lslga'`
should be
`if req.GET.get('lslga') == ''`.

I just submitted a pull request (#65) to hopefully fix that issue. My inspection tool is indeed drawing on top of the requested cutout, but my pull request should hopefully fix the issues with doing it directly on the cutout server.